CMakeLists.txt

```diff
@@ -9,8 +9,18 @@ file(GLOB MODELS_HEADERS torchvision/csrc/models/*.h)
 file(GLOB MODELS_SOURCES torchvision/csrc/models/*.h torchvision/csrc/models/*.cpp)
 
 add_library (${PROJECT_NAME} SHARED ${MODELS_SOURCES})
-target_link_libraries(${PROJECT_NAME} PUBLIC "${TORCH_LIBRARIES}")
+target_link_libraries(${PROJECT_NAME} PRIVATE "${TORCH_LIBRARIES}")
```

Comment:
I think that libtorch should be public dependency. According to [here](https://cmake.org/pipermail/cmake/2016-May/063400.html) ```When A links in B as *PRIVATE*, it is saying that A uses B in its
   implementation, but B is not used in any part of A's public API. Any code
   that makes calls into A would not need to refer directly to anything from
   B.```. Please tell me if I'm wrong here.

Reply:
Yes that would generally be true, however if the torch libraries are set as public dependencies, the generated config file expects them to also be exported: That  is, linking to Torch::Vision also links to all of pytorch. My idea was to model Vision as a sub-component of torch, forcing you to write:
```
target_link_libraries(my-target ${TORCH_LIBRARIES} Torch::Vision}
```
(maybe eventually replacing `${TORCH_LIBRARIES}` with  `Torch::LibTorch` or similar).
Additionally, the autogenerated cmake config file checks for  previous existance of the exported targets and
```
find_package(Torch REQUIRED)
find_package(TorchVision REQUIRED)
```
will result in target redefinition errors. That though can be fixed by providing a custom config file.

I am not bothered either way, and If you'd rather model TorchVision as an independent library (without the `Torch::` namespace) that carries along its depedencies I can add a custom cmake config file :)

Reply:
@bmanga I think it's better if libtorch is a public dependency since libtorch is not an implementation detail and is also part of the public API.

Reply:
Sounds good, I'll rework in the next couple of days :)